src/base/bittorrent/session.cpp

```diff
@@ -269,6 +288,8 @@ Session::Session(QObject *parent)
     , m_maxUploadsPerTorrent(BITTORRENT_SESSION_KEY("MaxUploadsPerTorrent"), -1, lowerLimited(0, -1))
     , m_isUTPEnabled(BITTORRENT_SESSION_KEY("uTPEnabled"), true)
     , m_isUTPRateLimited(BITTORRENT_SESSION_KEY("uTPRateLimited"), true)
+    , m_utpMixedMode(BITTORRENT_SESSION_KEY("uTPMixedMode"), m_isUTPEnabled, clampValue(0, 1))
```

Comment:
>As a user, how is one going to know the only valid values are FixedSlotsChoker, RateBasedChoker without source code at hand?

Oops... actually it needs a user manual, but we don't have it. Don't know what to say now... I would still use a textual representation.
>just got it working, but the new declared enum must accompany with a Q_ENUM_NS(...) macro which is only available on Qt >= 5.8. Also must add Q_NAMESPACE to namespace BitTorrent to support it.

Apparently, @evsh used something different... maybe, just enums (not enum classes). I think, we can use his code as base for cases of enumerable settings.